pkg/pgmodel/model/interface.go

```diff
@@ -27,7 +27,7 @@ var (
 
 // inserter is responsible for inserting label, series and data into the storage.
 type Inserter interface {
-	InsertNewData(rows map[string][]SamplesInfo) (uint64, error)
+	InsertNewData(rows DataWithTS) (uint64, error)
```

Comment:
why don't we just have this as two parameters instead?

Reply:
Having a different param for time may look trivial as the job of `InsertNewData` is to insert. So an abstraction here should be good. BTW, `DataWithTS` should be `Data` and then it makes proper sense.